utils/src/ast-grep/update-binding.ts

```diff
@@ -0,0 +1,351 @@
+	old?: string;
+	new?: string;
+};
+
+/**
+ * Update or remove a specific binding from an import or require statement.
+ *
+ * Analyzes the provided AST node to find and update a specific binding from destructured imports.
+ * If `newBinding` is provided in options, the binding will be replaced with the new name.
+ * If `newBinding` is not provided, the binding will be removed.
+ * If the binding is the only one in the statement and no replacement is provided, the entire import line is marked for removal.
+ *
+ * @param node - The AST node representing the import or require statement
+ * @param options - Optional configuration object
+ * @param options.old - The name of the binding to update or remove (e.g., "isNativeError")
+ * @param options.new - The new binding name to replace the old one. If not provided, the binding is removed.
+ * @returns An object containing either an edit operation or a line range to remove, or undefined if no binding found
+ *
+ * @example
+ * ```typescript
+ * // Given an import: const {types, isNativeError} = require("node:util")
+ * // And binding: "isNativeError", options: {newBinding: "isError"}
+ * // Returns: an edit object that transforms to: const {types, isError} = require("node:util")
+ * ```
+ *
+ * @example
+ * ```typescript
+ * // Given an import: const {types, isNativeError} = require("node:util")
+ * // And binding: "isNativeError", options: undefined
+ * // Returns: an edit object that transforms to: const {types} = require("node:util")
+ * ```
+ *
+ * @example
+ * ```typescript
+ * // Given an import: const {isNativeError} = require("node:util")
+ * // And binding: "isNativeError", options: {newBinding: "isError"}
+ * // Returns: an edit object that transforms to: const {isError} = require("node:util")
+ * ```
+ *
+ * @example
+ * ```typescript
+ * // Given an import: const {isNativeError} = require("node:util")
+ * // And binding: "isNativeError", options: undefined
+ * // Returns: {lineToRemove: Range} to remove the entire line
+ * ```
+ *
+ * @example
+ * ```typescript
+ * // Given an import: const util = require("node:util")
+ * // And binding: "isNativeError"
+ * // Returns: undefined (no destructured binding found)
+ * ```
+ */
+export function updateBinding(
+	node: SgNode<Js>,
+	options?: UpdateBindingOptions,
+): UpdateBindingReturnType {
+	const nodeKind = node.kind().toString();
+
+	const namespaceImport = node.find({
+		rule: {
+			any: [
+				{
+					kind: "identifier",
+					inside: {
+						kind: "variable_declarator",
+						// this `not rule` ensures that expressions like `require("something").NamedImport` are ignored
+						// because we only want the namespace to be returned here
+						not: {
+							has: {
+								field: "value",
+								kind: "member_expression",
+							},
+						},
+						inside: {
+							kind: "lexical_declaration",
+						},
+					},
+				},
+				{
+					kind: "identifier",
+					inside: {
+						kind: "import_clause",
+					},
+				},
```

Comment:
Wouldn't it be simpler to have two ast grep query ?

Reply:
Hmm, did you say something like it? 

```js
let identifiers = []
const importClauses = node.findAll(import_clause)
for(const importClause of importClauses) {
  const _identifiers = importClause.findAll(identifier)
  if(_identifiers.length) {
    identifiers = identifiers.concat(_identifiers)
  }
}
```